crates/did-utils/src/methods/peer/errors.rs

```diff
@@ -62,3 +62,11 @@ impl From<DIDPeerMethodError> for DIDResolutionError {
         }
     }
 }
+
+impl std::fmt::Display for DIDPeerMethodError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}", self) // Customize this to format the error as desired
+    }
+}
+
+impl std::error::Error for DIDPeerMethodError {}
```

Comment:
You are not supposed to manually implement the `Display` Trait. It's not recommended either. Instead you can derive the `Debug` Trait which will automatically implement `Display` or better use the `thiserror` crate which will automatically implement both `Error` and `Display`.
```rust
use thiserror::Error;

#[derive(Debug, Error)]
pub enum DIDPeerMethodError {
#[error("Invalid DID")]
InvalidDid,
...
}
```

Reply:
Okay

Reply:
![Screenshot from 2024-09-25 09-32-26](https://github.com/user-attachments/assets/ee35448a-f52e-44ae-b7e6-379c8cfe3c11)
I followed your suggestion to use `thiserror` instead of manually implementing the Display trait. However, I ran into an issue because the `crypto::errors::Error` type does not implement Display. Since `thiserror` relies on the Display trait for error formatting, I encountered the error:
```rust
error[E0599]: the method `as_display` exists for reference `&Error`, but its trait bounds were not satisfied
```
To resolve this, I would either need to manually implement the Display trait for `crypto::errors::Error` or refactor it to use `thiserror` as well, so that both `Display` and `Error` are automatically derived.

Reply:
@Hermann-Core What do you think?

Reply:
Alternatively, you could map the `DIDPeerMethodError` to a `DIDResolutionError`, as it already implements all the necessary `Trait` via `thisError`. Then, adjust the signature of the `expand` method in `did:peer` to match the `expand` method signature in `did:key`:
```rust
pub fn expand(&self, did: &str) -> Result<DIDDocument, DIDResolutionError> {
```
This way, there would be no need to handle different errors within the `resolve` method of the `LocalDidResolver`. I believe that the expansion (which in fact is the resolution) of DIDs should always return a `DIDResolutionError` in case of errors, regardless of the method used.

Reply:
Ok on it

Reply:
![Screenshot from 2024-09-25 12-41-04](https://github.com/user-attachments/assets/e902a456-4757-4765-8beb-921b0f59e3de)
 I think these solves the issue of mapping `DIDPeerMethodError` to `DIDResolutionError`

Reply:
@Hermann-Core Just pushed some current changes in respond to your review. Enable to check it and make sure it corresponds to what you are requesting.

Reply:
> I think these solves the issue of mapping `DIDPeerMethodError` to `DIDResolutionError`
Ok these have already been implemented. So only the signature of the `expand` method needs to be changed.